pkgs/applications/networking/cluster/talosctl/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, buildGo117Module, fetchFromGitHub }:
+
+buildGo117Module rec {
+  pname = "talosctl";
+  version = "0.14.1";
+
+  src = fetchFromGitHub {
+    owner = "talos-systems";
+    repo = "talos";
+    rev = "v${version}";
+    sha256 = "sha256-JeZ+Q6LTDJtoxfu4mJNc3wv3Y6OPcIUvgnozj9mWwLw=";
+  };
+
+  vendorSha256 = "sha256-ujbEWvcNJJOUegVgAGEPwYF02TiqD1lZELvqc/Gmb4A=";
+
+  # look for GO_LDFLAGS getting set in the Makefile
```

Comment:
stupid question: is there a reason not to use the Makefile directly?

Reply:
I don't know how to compose the `buildGo*Module` dependency fetching tooling with that makefile, and there's a bunch of targets in the Makefile using shebangs that'd need to be patched.

I'd rather prefer using `buildGo*Module`.

Reply:
You can use `buildGoModule` and than override the `buildPhase`. We still want `fixupPhase` since it removes references on the go compiler. This is what we do for a number of other packages.

Reply:
Nope. I looked at the makefile, even a make `local-talosctl` will invoke `target-talosctl`, which will invoke `BUILD`, which is `docker buildx build`.

It'll probably be more effort to maintain a Makefile patch that undoes all the docker wrapping to use the go compiler in $PATH, inside the sandbox.

Reply:
fair.